openai/src/commonMain/kotlin/com/xebia/functional/xef/auto/llm/openai/MockAIClient.kt

```diff
@@ -28,6 +29,9 @@ class MockOpenAIClient(
   private val chatCompletion: (ChatCompletionRequest) -> ChatCompletionResponse = {
```

Comment:
Should we maybe split these methods in a different interface? I see more and more that a client implements the interface only partially.

Reply:
They've already been split into its own interface in `main` inside `Chat`, `ChatWithFunctions` etc. The remaining place where it implements all of them like this is the MockClient. The `AIClient` interface in the main is already unused and should be removed. I'll push a commit to this PR to remove it.